### PR TITLE
[STACK-168] Correct duplicate argument issue in listener update

### PR DESCRIPTION
**Pull request by mdurrant-b3**
Corrects issue reported by customer when `protocol` is specified in virtual port expressions.

### CODE REVIEW

File: a10_neutron_lbaas/v2/handler_listener.py [hunk 1]
```diff
@@ -177,6 +179,11 @@ def _set(self, set_method, c, context, listener):
         if hasattr(listener, 'aflex'):
             template_args["aflex-scripts"] = listener.aflex
 
+        if "protocol" in vport_defaults:
+            # import pdb; pdb.set_trace()
```
Comment (hthompson6) [inline a10_neutron_lbaas/v2/handler_listener.py]: Should be removed completely.

File: a10_neutron_lbaas/v2/handler_listener.py [hunk 2]
```diff
@@ -177,6 +179,11 @@ def _set(self, set_method, c, context, listener):
         if hasattr(listener, 'aflex'):
             template_args["aflex-scripts"] = listener.aflex
 
+        if "protocol" in vport_defaults:
```
Comment (hthompson6) [inline a10_neutron_lbaas/v2/handler_listener.py]: Shouldn't this be done higher up as the protocol, which will soon be set, effects above logic?

Comment (mdurrant-b3) [reply a10_neutron_lbaas/v2/handler_listener.py]: This ensures the override from vport expressions applies so we still apply correct HTTPS logic.

File: a10_neutron_lbaas/v2/handler_listener.py (hunk 1 above)
Comment (mdurrant-b3) [reply a10_neutron_lbaas/v2/handler_listener.py]: Thanks

**Comment by mdurrant-b3** [comment]
The pypy tests appear to be failing due to a bad pypy ver or pypy-incompatible tox ver being pulled by Travis.

**Comment by mdurrant-b3** [comment]
check-ci

**Comment by mdurrant-b3** [comment]
check

**Pull request merged**